### ISSUE TITLE
Assertion error when using blog plugin

### ISSUE THREAD

**Issue by jiegec** [squidfunk/mkdocs-material#5880]
### Bug description

After upgrading to mkdocs-material 9.2.1, an assertion error occurred:

```
ERROR   -  Error building page 'index.md':
Traceback (most recent call last):
  File "/Users/macbookair/Library/Caches/pypoetry/virtualenvs/mkdocs-bug-v9bxKRa0-py3.11/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/Users/macbookair/Library/Caches/pypoetry/virtualenvs/mkdocs-bug-v9bxKRa0-py3.11/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/macbookair/Library/Caches/pypoetry/virtualenvs/mkdocs-bug-v9bxKRa0-py3.11/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/Users/macbookair/Library/Caches/pypoetry/virtualenvs/mkdocs-bug-v9bxKRa0-py3.11/lib/python3.11/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/macbookair/Library/Caches/pypoetry/virtualenvs/mkdocs-bug-v9bxKRa0-py3.11/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/macbookair/Library/Caches/pypoetry/virtualenvs/mkdocs-bug-v9bxKRa0-py3.11/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/macbookair/Library/Caches/pypoetry/virtualenvs/mkdocs-bug-v9bxKRa0-py3.11/lib/python3.11/site-packages/mkdocs/__main__.py", line 286, in build_command
    build.build(cfg, dirty=not clean)
  File "/Users/macbookair/Library/Caches/pypoetry/virtualenvs/mkdocs-bug-v9bxKRa0-py3.11/lib/python3.11/site-packages/mkdocs/commands/build.py", line 349, in build
    _build_page(
  File "/Users/macbookair/Library/Caches/pypoetry/virtualenvs/mkdocs-bug-v9bxKRa0-py3.11/lib/python3.11/site-packages/mkdocs/commands/build.py", line 222, in _build_page
    context = config.plugins.on_page_context(context, page=page, config=config, nav=nav)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/macbookair/Library/Caches/pypoetry/virtualenvs/mkdocs-bug-v9bxKRa0-py3.11/lib/python3.11/site-packages/mkdocs/plugins.py", line 583, in on_page_context
    return self.run_event('page_context', context, page=page, config=config, nav=nav)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/macbookair/Library/Caches/pypoetry/virtualenvs/mkdocs-bug-v9bxKRa0-py3.11/lib/python3.11/site-packages/mkdocs/plugins.py", line 507, in run_event
    result = method(item, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/macbookair/Library/Caches/pypoetry/virtualenvs/mkdocs-bug-v9bxKRa0-py3.11/lib/python3.11/site-packages/material/plugins/blog/plugin.py", line 287, in on_page_context
    assert isinstance(main.parent, Section)
AssertionError
```

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
- [Setting up a blog](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/)

### Reproduction

[assertion-failed.zip](https://github.com/squidfunk/mkdocs-material/files/12403848/assertion-failed.zip)

### Steps to reproduce

1. unzip assertion-failed.zip
2. remove info plugin
3. run `mkdocs build`

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting! Fixed in 62d4a1606.

**Comment by squidfunk** [comment]
Released as part of [9.2.2](https://squidfunk.github.io/mkdocs-material/changelog/#9.2.2).

**Comment by ianderrington** [comment]
As a note: I got this bug (I'm using the awesome-pages plugin so maybe that is why?) and I am on `9.5.3+insiders-4.49.0` I just commented out the assertion and it works fine.
